### ISSUE TITLE
Serde support?

### ISSUE THREAD

**Issue by fxtressia** [QEDK/configparser-rs#33]
Hi! Can we have serde support in configparser?

**Comment by QEDK** [comment]
> Hi! Can we have serde support in configparser?

The question isn't _can_ but why? If you have a use-case for it, I think we can think about possibilities as to why we should be adding it as a feature.

**Comment by lovelyyfiaaa** [comment]
> > Hi! Can we have serde support in configparser?
> 
> The question isn't _can_ but why? If you have a use-case for it, I think we can think about possibilities as to why we should be adding it as a feature.

Well, I wanted to have the data be represented as hard-typed structs but doing that without Serde seems to be verbosyy ^^

https://github.com/waylovely-project/flara-shop/blob/eeefb5f6f3c9a23e0b21618eac4f2c62acfbae33/flarapak/src/schemas/remote.rs

**Comment by hansl** [comment]
Yes. Hard typed structures is also a use case for us. This library is actually very hard to use and very verbose without some kind of derive based structure deserialization. Serde is just the easiest framework to implement this, and already a pseudo-standard in the Rust ecosystem.

What I want is something like:

```rust
#[derive(Deserialize)] // Could be from configparser, but would prefer serde.
pub struct Config {
  some_config: String,
  some_other_config: Option<u32>,
}

impl Config {
  pub fn load(path: impl AsRef<Path>) -> Result<Self, HereBeErrType> {
    let str = std::fs::read_to_string(path)?;
    configparser::parse::<Self>(str)
  }
}
```

**Comment by QEDK** [comment]
> Yes. Hard typed structures is also a use case for us. This library is actually very hard to use and very verbose without some kind of derive based structure deserialization. Serde is just the easiest framework to implement this, and already a pseudo-standard in the Rust ecosystem.
> 
> What I want is something like:
> 
> ```rust
> #[derive(Deserialize)] // Could be from configparser, but would prefer serde.
> pub struct Config {
>   pub some_config: String,
>   pub some_other_config: Option<u32>,
> }
> 
> impl Config {
>   pub fn load(path: impl AsRef<Path>) -> Result<Self, HereBeErrType> {
>     let str = std::fs::read_to_string(path)?;
>     configparser::parse::<Self>(&str)
>   }
> }
> ```
The work is now underway, I think as the wider ecosystem expands to use `serde`, it only makes sense to provide native support.

**Comment by QEDK** [comment]
An easy way might be to use `serde_json`, parse our unparsed string and
return the generic type? It is inefficient but doesn't require rebuilding
the wheel.
~~

**Comment by BLumia** [comment]
> The work is now underway

Any updates on serde support? Is there a related Pull Request so we could track the progress?

**Comment by QEDK** [comment]
@BLumia @togetherwithasteria this has been implemented on the `main` branch, you're invited to test it out before I release it later this week.

**Comment by QEDK** [comment]
This was fixed in an earlier version.
